pkg/gops/cell_test.go

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+//go:build !privileged_tests
+
+package gops
+
+import (
+	"io/ioutil"
+	"net"
+	"strings"
+	"testing"
+
+	gopsSignal "github.com/google/gops/signal"
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"github.com/cilium/cilium/pkg/hive"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+func TestGopsCell(t *testing.T) {
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	viper := viper.New()
+
+	hive := hive.New(
+		viper, flags,
+		Cell,
+	)
+	flags.Set(option.GopsPort, "15151")
+
+	app, err := hive.TestApp(t)
+	if err != nil {
+		t.Fatalf("TestApp(): %s", err)
+	}
+
+	app.RequireStart()
+
+	conn, err := net.Dial("tcp", "127.0.0.1:15151")
+	if err != nil {
+		t.Fatalf("Dial(:15151): %s", err)
+	}
+	defer conn.Close()
```

Comment:
The gops agent starts listening on a separate goroutine: https://github.com/cilium/cilium/blob/1fd3c08d0a186866602cdef02eddf78a93051c80/vendor/github.com/google/gops/agent/agent.go#L123

So I think it would be possible that the fx framework completes the app start phase while the gops agent is not yet ready to actually listen.
May be worth adding a retry here? However, I was unable to reproduce it locally.

Reply:
Good catch, definitely will add a retry and run this a bunch of times to make sure I won't introduce a flaky test.
OTOH, I only added this test to give an example of how a Cell would be tested, so might consider just dropping this instead as this doesn't add much value.